src/Microsoft.AspNet.Http/Features/QueryFeature.cs

```diff
@@ -41,11 +41,23 @@ void IFeatureCache.CheckFeaturesRevision()
         {
             if (_cachedFeaturesRevision != _features.Revision)
             {
-                _request = null;
-                _cachedFeaturesRevision = _features.Revision;
+                ResetFeatures();
             }
         }
 
+        public void UpdateFeatures(IFeatureCollection features)
```

Comment:
I think you're conflating two types of objects, Managers and Features. The Managers have UpdateFeatures, where the Features : IFeatureCache with CheckFeaturesRevision. CheckFeaturesRevision never appears to be called on the Manager objects.